packages/ui/src/Utilities.tsx

```
@@ -162,3 +164,88 @@ export const union =
 export const isNative = (): boolean => {
   return ["android", "ios"].includes(Platform.OS);
 };
+
+// Find more information about the address component types here: https://developers.google.com/maps/documentation/javascript/place-autocomplete
+export type AddressComponentType = {
+  long_name: string;
+  short_name: string;
+  types: string[];
+};
+
+export const findAddressComponent = (components: AddressComponentType[], type: string): string => {
+  return (
+    components.find((component: AddressComponentType) => component.types.includes(type))
+      ?.long_name ?? ""
+  );
+};
+
+export const processAddressComponents = (addressComponents: AddressComponentType[] | undefined) => {
+  if (!addressComponents || addressComponents.length === 0) {
+    return {
+      address1: "",
+      city: "",
+      state: "",
+      zipcode: "",
+      countyName: "",
+    };
+  }
+  const streetNumber = findAddressComponent(addressComponents, "street_number");
+  const streetName = findAddressComponent(addressComponents, "route");
+  const city = findAddressComponent(addressComponents, "locality");
+  const state = findAddressComponent(addressComponents, "administrative_area_level_1");
+  const zipcode = findAddressComponent(addressComponents, "postal_code");
+  const countyName = findAddressComponent(addressComponents, "administrative_area_level_2");
+
+  if (state && countyName) {
+    const countyCode = formattedCountyCode(state, countyName);
+    return {
+      address1: `${streetNumber} ${streetName}`.trim(),
+      city,
+      state,
+      zipcode,
+      countyName,
+      countyCode,
+    };
+  }
+  return {
+    address1: `${streetNumber} ${streetName}`.trim(),
+    city,
+    state,
+    zipcode,
+    countyName,
+  };
+};
+
+// Google does not provide a way to validate API keys, so we have to do it ourselves
+export const isValidGoogleApiKey = (apiKey: string): boolean => {
```

Can we add some console.warns here instead of silently failing?